diff --git a/src/codec/id.rs b/src/codec/id.rs
index 22d27a7..9ae9066 100644
--- a/src/codec/id.rs
+++ b/src/codec/id.rs
@@ -1139,6 +1139,7 @@ impl From<AVCodecID> for Id {
             AV_CODEC_ID_HCA => Id::HCA,
             #[cfg(feature = "ffmpeg_4_3")]
             AV_CODEC_ID_EPG => Id::EPG,
+            _ => Id::H264,
         }
     }
 }
diff --git a/src/codec/packet/side_data.rs b/src/codec/packet/side_data.rs
index 3322f2c..c19b69e 100644
--- a/src/codec/packet/side_data.rs
+++ b/src/codec/packet/side_data.rs
@@ -94,6 +94,7 @@ impl From<AVPacketSideDataType> for Type {
             AV_PKT_DATA_ICC_PROFILE => Type::ICC_PROFILE,
             #[cfg(feature = "ffmpeg_4_3")]
             AV_PKT_DATA_DOVI_CONF => Type::DOVI_CONF,
+            _ => Type::Palette,
         }
     }
 }
diff --git a/src/util/format/pixel.rs b/src/util/format/pixel.rs
index 96a9072..9b19e5f 100644
--- a/src/util/format/pixel.rs
+++ b/src/util/format/pixel.rs
@@ -687,6 +687,7 @@ impl From<AVPixelFormat> for Pixel {
             AV_PIX_FMT_RPI4_8 => Pixel::RPI4_8,
             #[cfg(feature = "rpi")]
             AV_PIX_FMT_RPI4_10 => Pixel::RPI4_10,
+            _ => Pixel::RGBA,
         }
     }
 }
diff --git a/src/util/frame/side_data.rs b/src/util/frame/side_data.rs
index 14591ae..04b14cf 100644
--- a/src/util/frame/side_data.rs
+++ b/src/util/frame/side_data.rs
@@ -91,6 +91,7 @@ impl From<AVFrameSideDataType> for Type {
 
             #[cfg(feature = "ffmpeg_4_3")]
             AV_FRAME_DATA_VIDEO_ENC_PARAMS => Type::VIDEO_ENC_PARAMS,
+            _ => Type::VIDEO_ENC_PARAMS,
         }
     }
 }
